identifier/backends/ldap/ldap.go

```diff
@@ -561,6 +568,20 @@ func (b *LDAPIdentifierBackend) connect(parentCtx context.Context) (*ldap.Conn,
 	return l, nil
 }
 
+// connect establishes a global connection that is used for all requests except the user Bind to check the password in the Logon call.
+func (b *LDAPIdentifierBackend) connect(parentCtx context.Context) (*ldap.Conn, error) {
+	b.connLock.Lock()
+	defer b.connLock.Unlock()
+
+	if b.conn != nil && !b.conn.IsClosing() {
```

Comment:
This does not really know if the connection actually still works and might result in error and thus logon failure when getting used. Does the ldap connection do anything (like in the background) to ensure that does not happen? Normally that would be the job of the pool before giving the connection to the caller.

Reply:
Hm, the connection closes itself only when it cannot read a response. We could port the reconnect wrapper from https://github.com/cs3org/reva/blob/edge/pkg/utils/ldap/reconnect.go

Reply:
I think this needs to be solved before this can be merged. One way or another the connection needs to be "checked" if it is still usable (or some retry logic).